crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -161,12 +164,21 @@ public Response insert(InsertionRequest req) {
                 ctx.getHookManager().callQueuedHooks();
             }
         } catch (Error e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
+            }
             response.getErrors().add(e);
             response.setStatus(OperationStatus.ERROR);
         } catch (Exception e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
```

Comment:
Related to getting more context, there is another option we should consider: accepting a RequestMetrics.Context in the Mediator methods like find, insert, etc.

This could get us the best of both worlds: broad coverage of the lifecycle of a request, but also coverage of the innards (like within bulk requests), at the cost of some complexity.

However, I think it might be worth it. For example, right now, if a request fails to parse, we won't get any metrics on that, because it never makes it to the Mediator. However, if we started tracking before we got to the Mediator, we could capture that.

Within bulk requests, then, it would do both: use the provided context for the bulk request, but also create new contexts to track the individual requests within it.

What do you think?